### PR TITLE
Fixes for login

### PR DESCRIPTION
**Pull request by craffit**
- Fixes a small bug in the httpactionadapter
- Adds callback functionality for indirect clients

### CODE REVIEW

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 1]
```diff
@@ -66,41 +68,73 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage)(implicit
   import AkkaHttpSecurity._
   // TODO: At some point this object should contain the SessionStore (when we implement that)
```
Comment (lmreis89) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala]: Remove this comment, session store is in place.

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 2]
```diff
@@ -66,41 +68,73 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage)(implicit
...
-             ): Directive1[AuthenticatedRequest] =
-    new Directive1[AuthenticatedRequest] {
-      override def tapply(innerRoute: Tuple1[AuthenticatedRequest] => Route): Route = { ctx =>
+  private[pac4j] val callbackLogic: CallbackLogic[Future[RouteResult], AkkaHttpWebContext] =
+    Option(config.getCallbackLogic) match {
+      case Some(v) => v.asInstanceOf[CallbackLogic[Future[RouteResult], AkkaHttpWebContext]]
+      case None => new DefaultCallbackLogic[Future[RouteResult], AkkaHttpWebContext]
+    }
+
+  private[pac4j] def withContext[A](enforceFormEncoding: Boolean): Directive1[AkkaHttpWebContext] =
```
Comment (lmreis89) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala]: Please add comments to this private route

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 3]
```diff
@@ -66,41 +68,73 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage)(implicit
...
+  /**
+    * Authenticate using the provided pac4j configuration. Delivers an AuthenticationRequest which can be used for further authorization
+    * this does not apply any authorization ofr filtering.
+    */
+  def withAuthentication(
+                          clients: String = null /* Default null, meaning all defined clients */ ,
+                          multiProfile: Boolean = true,
+                          enforceFormEncoding: Boolean = false, //Force form parameters to be passed for authentication or the request fails
+                        ): Directive1[AuthenticatedRequest] =
+    withContext(enforceFormEncoding).flatMap { akkaWebContext =>
```
Comment (lmreis89) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala]: cool stuff, looks cleaner.

File: src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala [hunk 4]
```diff
@@ -21,6 +21,8 @@ object AkkaHttpActionAdapter extends HttpActionAdapter[Future[RouteResult], Akka
         HttpResponse(Created)
       case HttpConstants.FORBIDDEN =>
         HttpResponse(Forbidden)
+      case HttpConstants.TEMP_REDIRECT =>
```
Comment (lmreis89) [inline src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala]: Does the play integration do anything with the response headers/cookies in this case?

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala (hunk 1 above)
Comment (craffit) [reply src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala]: done

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala (hunk 2 above)
Comment (craffit) [reply src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala]: done

File: src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala (hunk 4 above)
Comment (craffit) [reply src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala]: play uses the location header to set the redirect location. we just use the basic header mechanism to apply the location header to the response (the AkkaHttpSecurity.applyHeadersAndCookieToResponse)

**Pull request merged**